gh-107877: Update logging levels reference table with usage criteria. (#107894)
python/cpython: Doc/library/logging.rst

Reviewer: You could consider converting the table into a `list-table`, like we use in the typing docs here: https://github.com/python/cpython/blob/2e1f688fe0f0a612e54c09f5a7027a834dd8b8d5/Doc/library/typing.rst?plain=1#L3693

It might make it easier to maintain it

Author: Thanks for the suggestion - perhaps not worth changing it again now for this specific instance, but I'll bear in mind when creating tables in future!

with the same numeric value, it overwrites the predefined value; the predefined
name is lost.

+-----------------------+---------------+-------------------------------------+
| Level                 | Numeric value | What it means / When to use it      |
+=======================+===============+=====================================+
| .. py:data:: NOTSET   | 0             | When set on a logger, indicates that|
|                       |               | ancestor loggers are to be consulted|

> |                       |               | to determine the effective level. If|

Reviewer: Duplicate 'if'

```suggestion
|                       |               | to determine the effective level.   |
```

|                       |               | If that still resolves to           |

> |                       |               | :const:`NOTSET`, then all events are|
> |                       |               | logged. When set on a handler, all  |
> |                       |               | events are handled.                 |

Reviewer: The markup for NOTSET here would cause it to add a link which would just take you to where you already are, which would be annoying. Better to use `!` to suppress the link:

```suggestion
|                       |               | :const:`!NOTSET`, then all events   |
|                       |               | are logged. When set on a handler,  |
|                       |               | all events are handled.             |
```

+-----------------------+---------------+-------------------------------------+

> | .. py:data:: DEBUG    | 10            | Detailed information, typically of  |
> |                       |               | interest only when diagnosing       |
> |                       |               | problems, usually just for          |
> |                       |               | developers.                         |

Reviewer: Slightly more concise:

```suggestion
| .. py:data:: DEBUG    | 10            | Detailed information, typically only|
|                       |               | of interest to a developer trying to|
|                       |               | diagnose a problem.                 |
```

+-----------------------+---------------+-------------------------------------+
| .. py:data:: INFO     | 20            | Confirmation that things are working|
|                       |               | as expected.                        |
+-----------------------+---------------+-------------------------------------+
| .. py:data:: WARNING  | 30            | An indication that something        |

> |                       |               | unexpected happened, or indicative  |
> |                       |               | of a problem in the near future     |
> |                       |               | (e.g. 'disk space low'). The        |

Reviewer: ```suggestion
|                       |               | unexpected happened, or that a      |
|                       |               | problem might occur in the near     |
|                       |               | future (e.g. 'disk space low'). The |
```
